src/net.cpp

```diff
@@ -2278,6 +2283,17 @@ void ThreadMessageHandler()
 
         bool fSleep = true;
 
+        if ((std::chrono::duration_cast<std::chrono::microseconds>(
+                 std::chrono::high_resolution_clock::now() - lastMempoolSync)
+                    .count() > MEMPOOLSYNC_FREQ_US) &&
+            vNodesCopy.size() > 0)
+        {
+            // select node from whom to request mempool sync
+            CNode *syncPeer = SelectMempoolSyncPeer(vNodesCopy);
+            if (!(syncPeer == nullptr))
+                requester.RequestMempoolSync(syncPeer);
```

Comment:
I don't think we need to start requesting mempool sync until our own chain is almost syncd and indeed we don't even request transactions until we're out of IBD anway.

-if (!(syncPeer == nullptr))
+if (!(syncPeer == nullptr) && IsChainNearlySyncd())